File: java/eyes.sdk.core/src/main/java/com/applitools/eyes/MatchLevel.java
package com.applitools.eyes;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonValue;
import com.google.gson.annotations.SerializedName;

/**
 * The extent in which two images match (or are expected to match).
 */
public enum MatchLevel {
    /**
     * Images do not necessarily match.
     */
    NONE("None"),

    /**
     * Images have the same layout.
     */
    LAYOUT("Layout"),

    /**
     * Images have the same layout.
     */
    LAYOUT2("Layout2"),

    /**
     * Images have the same outline.
     */
    @Deprecated
    CONTENT("Content"),

    /**
     * Images have the same outline.
     * Formerly known as Content.
     */
    IGNORE_COLORS("IgnoreColors"),

    /**
     * Images are nearly identical.
     */
    STRICT("Strict"),

    /**
     * Images are identical.
     */
    EXACT("Exact");

    private final String name;

    MatchLevel(String name) { this.name = name; }

    public String getName() {
        if (name.equals("Content"))
            System.out.println("The \"Content\" match level value has been deprecated, use \"IgnoreColors\" instead.");

        return name;
    }

    @JsonCreator
    public static MatchLevel fromString(String key) {
        return key == null ? null : MatchLevel.valueOf(key.toUpperCase());
    }
}

File: dotnet/Eyes.Images.DotNet/Enums/MatchLevel.cs
using System;

namespace Applitools
{
    /// <summary>
    /// The extent in which two images match (or are expected to match).
    /// </summary>
    public enum MatchLevel
    {
        /// <summary>
        /// Images do not necessarily match.
        /// </summary>
        None,

        /// <summary>
        /// Images have the same layout (compared using the latest layout comparison engine).
        /// </summary>
        Layout,

        /// <summary>
        /// Images have the same layout (compared using layout comparison engine v1).
        /// </summary>
        Layout1,

        /// <summary>
        /// Images have the same layout (compared using layout comparison engine v2).
        /// </summary>
        Layout2,

        /// <summary>
        /// Images have the same content.
        /// </summary>
        [Obsolete("Use IgnoreColors")]
        Content = 4,

        /// <summary>
        /// Images have the same  outline.
        /// Formerly known as Content.
        /// </summary>
        IgnoreColors = 4,

        /// <summary>
        /// Images are nearly identical.
        /// </summary>
        Strict,

        /// <summary>
        /// Images are identical.
        /// </summary>
        Exact
    }
}
